packages/atomic/src/decorators/light-dom.spec.ts

```diff
@@ -0,0 +1,88 @@
+import {fixture} from '@/vitest-utils/testing-helpers/fixture';
+import {html, LitElement, unsafeCSS} from 'lit';
+import {injectStylesForNoShadowDOM} from './light-dom';
+
+describe('injectStylesForNoShadowDOM', () => {
+  const styles = 'body { background-color: red; }';
+
+  @injectStylesForNoShadowDOM
+  class StyledElement extends LitElement {
+    static styles = unsafeCSS(styles);
+    render() {
+      return html`<div>children element</div>`;
+    }
+  }
+
+  class LightDomParentElement extends HTMLElement {
+    render() {
+      return html`<slot></slot>`;
+    }
+  }
+
+  class ShadownDomParentElement extends HTMLElement {
+    constructor() {
+      super();
+      this.attachShadow({mode: 'open'});
+    }
+
+    connectedCallback() {
+      const fragment = document.createDocumentFragment();
+      while (this.childNodes.length > 0) {
+        fragment.appendChild(this.childNodes[0]);
+      }
+      this.shadowRoot!.appendChild(fragment);
+    }
+  }
+
+  customElements.define('light-dom-parent-element', LightDomParentElement);
+  customElements.define('shadow-dom-parent-element', ShadownDomParentElement);
+  customElements.define('styled-element', StyledElement);
+
+  describe('when the element is added in a shadow dom', () => {
```

Comment:
For both light and shadow DOM, we could also test that:

- if there's no stylesheet, nothing happens
- if the parent's adopted stylesheets already includes the stylesheet, nothing happens